subsys/bluetooth/host/smp.c

```diff
@@ -1472,6 +1472,14 @@ static void smp_pairing_complete(struct bt_smp *smp, u8_t status)
 {
 	BT_DBG("status 0x%x", status);
 
+	if (bt_auth->pairing_completed) {
+		bt_pairing_result_t pairing_ret = status == 0 ?
+			BT_PAIRING_STATUS_BONDED :
```

Comment:
I'm not sure you understand what bonded means. There's a flag in the smp struct for that. "bonded" means you're paired and intend to store the keys persistently. "paired" is a superset of "bonded" which doesn't guarantee that the keys will be stored persistently. In the Zephyr case when pairing, but not bonding, the keys are not stored beyond the connection that creates them, i.e. they're lost once you disconnect. So mapping "status == 0" to "bonded" is not right.

Reply:
Thanks for clarifying! Where can the function "amp_pairing_complete" know that the pairing is "bonded" from its argument? The "bonded" status has been stored somewhere?